examples/kubernetes/README.md

```diff
@@ -1,170 +1,215 @@
+
+```
+kubectl get svc
 ```
-NAME                                    READY     STATUS    RESTARTS   AGE
-kube-dns-v20-1485703853-mqbsq           3/3       Running   0          1m
-kube-dns-v20-1485703853-q742f           3/3       Running   0          1m
 ```
+$ kubectl get svc
+NAME           CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
+guestbook      fd03::ee8c   <none>        3000/TCP   4m
+kubernetes     fd03::1      <none>        443/TCP    58m
+redis-master   fd03::2e03   <none>        6379/TCP   5m
+redis-slave    fd03::b1e7   <none>        6379/TCP   5m
+```
+
+### Create ingress rules
 
-*Wait until all pods are in running state*
+You now have a set of pods and services inside your cluster. That isn't too much
+helpful unless they are exposed outside of the cluster.
 
-## Run guestbook service
+TODO: rewrite ingress-name
 
-Now that we have a proper kubernetes cluster ready for our services we can run a simple service in our cluster:
 ```
-kubectl create -f https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/deployments/guestbook/1-redis-master-controller.json
-kubectl create -f https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/deployments/guestbook/2-redis-master-service.json
-kubectl create -f https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/deployments/guestbook/3-redis-slave-controller.json
-kubectl create -f https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/deployments/guestbook/4-redis-slave-service.json
-kubectl create -f https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/deployments/guestbook/5-guestbook-controller.json
-kubectl create -f https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/deployments/guestbook/6-guestbook-service.json
+kubectl -f create examples/kubernetes/deployments/guestbook/ingress/ingress.yml
 ```
-Check the deployment
 ```
-kubectl get pods -o wide
+ingress "test-ingress" created
 ```
+Verify if the ingress controller set up an address for the given ingress rule:
 ```
-NAME                 READY     STATUS    RESTARTS   AGE       IP             NODE
-guestbook-dxg09      1/1       Running   0          3m        10.1.16.218    cilium-master
-redis-master-rpxm0   1/1       Running   0          3m        10.1.100.129   cilium-master
-redis-slave-3vv08    1/1       Running   0          3m        10.1.15.138    cilium-master
+kubectl get ingresses
 ```
-*Wait until all pods are in running state*
+```
+NAME           HOSTS     ADDRESS   PORTS     AGE
+test-ingress   *                   80        6s
+```
+
+As you may noticed, cilium as set up the public address of `cilium-k8s-master`
+as the ingress address. This will redirect all traffic that enters that IP with
+port 80 to the `guestbook` service running on `cilium-k8s-node-2`. The response
+packets from `guestbook` service will go directly to the client without touching
+the loadbalancer node (`cilium-k8s-master`).
```

Comment:
This loadbalancing mode is referred to as direct server return and decreases the system load on the load balancers as reverse translation is distributed across all backend nodes. This is in particular beneficial on asymmetric bandwidth workloads with small requests and potentially large responses such as media streaming.